app/controllers/accounts_controller.rb

```diff
@@ -26,4 +26,11 @@ def create
   helper_method memoize def account
     Account.visible_for(current_user).find(ps.fetch(:id))
   end
+
+  helper_method memoize def read_only?(id)
+    return false if current_user.account.id == account.parent.id ||
+                    AccountShare.accepted.for(current_user).pluck(:account_id).include?(id)
```

Comment:
Looks like we need an authorization gem, we also need to check the actions inside controllers. hiding the buttons on UI is not enough.

We can handle that in a further PR.

Reply:
UPD: I think we even don't need to check the actions on back-end. They should work now correctly. So feel free to ignore this comment.